infra/staging/main.tf

```diff
@@ -152,6 +157,107 @@ resource "google_storage_bucket_iam_member" "frontend_bucket_viewer" {
   member = "allUsers"
 }
 
+#######################################
+## Load balancer for frontend bucket ##
+#######################################
+
+// Following the guide here: https://cloud.google.com/cdn/docs/setting-up-cdn-with-bucket
+
+// Enable Compute Engine API to setup an IP address
+resource "google_project_service" "compute_engine_api" {
+  project = data.google_project.project.project_id
+  service = "compute.googleapis.com"
+}
+
+// Setup IP address for load balancer
+resource "google_compute_global_address" "call_home_frontend_ip" {
+  project = data.google_project.project.project_id
+  name = "call-home-frontend-ip"
+}
+
+// Backend bucket with CDN policy with default ttl settings
+resource "google_compute_backend_bucket" "call_home_frontend_lb" {
+  name = "call-home-frontend-lb"
+  description = "Load balancer for the frontend GCS bucket"
+  project = data.google_project.project.project_id
+  bucket_name = google_storage_bucket.call_home_frontend.name
+  enable_cdn = true
+
+  cdn_policy {
+    cache_mode = "CACHE_ALL_STATIC"
+    client_ttl = 3600
+    default_ttl = 3600
+    max_ttl = 86400
+    negative_caching = true
+    serve_while_stale = 86400
+  }
+}
+
+// URL map for the backend bucket
+resource "google_compute_url_map" "call_home_lb" {
+  name = "call-home-lb"
+  project = data.google_project.project.project_id
+  default_service = google_compute_backend_bucket.call_home_frontend_lb.id
+}
+
+// Setup SSL certificate to enable HTTPS for load balancer
+resource "google_compute_managed_ssl_certificate" "call_home_frontend_lb_ssl" {
+  name = "call-home-frontend-lb-ssl"
+  project = data.google_project.project.project_id
+
+  managed {
+    domains = [var.website_domain_name]
+  }
+}
+
+// HTTPS proxy
+resource "google_compute_target_https_proxy" "call_home_frontend_lb_proxy" {
+  name = "call-home-frontend-lb-proxy"
+  project = data.google_project.project.project_id
+  url_map = google_compute_url_map.call_home_lb.id
+
+  ssl_certificates = [
+    google_compute_managed_ssl_certificate.call_home_frontend_lb_ssl.name
+  ]
+}
+
+// Forwarding rule
+resource "google_compute_global_forwarding_rule" "call_home_frontend_lb_forwarding" {
+  name = "call-home-frontend-lb-forwarding"
+  project = data.google_project.project.project_id
+  ip_protocol = "TCP"
+  load_balancing_scheme = "EXTERNAL"
+  port_range = "443"
+  target = google_compute_target_https_proxy.call_home_frontend_lb_proxy.id
+  ip_address = google_compute_global_address.call_home_frontend_ip.id
+}
+
+// The following resources create a partial load balancer
+// for redirecting HTTP to HTTPS
+resource "google_compute_url_map" "call_home_lb_http_redirect" {
+  name = "call-home-lb-http-redirect"
+  project = data.google_project.project.project_id
+
+  default_url_redirect {
+    strip_query            = false
+    https_redirect         = true
```

Comment:
Interesting. So this is a simple HTTP proxy that redirects all requests to HTTPS, preserving the route. As a result, the user would get sent to the HTTPS proxy and all of the real routing can commence.

Not sure if this is elegant or Rube Goldberg-y, but it is simple, so I think this works well for us :) Maybe one day we'll notice a way to do this with one proxy, and maybe that'll save us some money, but we'll worry about that when the time comes.